pkg/distributor/distributor.go

```diff
@@ -448,6 +454,28 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	var validationErrors util.GroupedErrors
 	validationContext := d.validator.getValidationContextForTime(time.Now(), tenantID)
 
+	shardStreamsCfg := d.validator.Limits.ShardStreams(tenantID)
+	maybeShardByRate := func(stream logproto.Stream, pushSize int) {
+		if shardStreamsCfg.Enabled {
+			streams = append(streams, d.shardStream(stream, pushSize, tenantID)...)
+		} else {
+			streams = append(streams, KeyedStream{
+				HashKey: lokiring.TokenFor(tenantID, stream.Labels),
+				Stream:  stream,
+			})
+		}
+	}
+	maybeShardByTime := func(stream logproto.Stream, labels labels.Labels, pushSize int) {
+		if shardStreamsCfg.TimeShardingEnabled {
```

Comment:
can be inverted to remove the else block. example:
```suggestion
		if !shardStreamsCfg.TimeShardingEnabled {
		    maybeShardByRate(stream, pushSize)
		    return
		}
```

Reply:
Resolved by https://github.com/grafana/loki/pull/14711/commits/ad1c0b2ad882121da20acda5e50db3833ac0a39d